pkg/model/keycloak_deployment.go

```diff
@@ -174,10 +174,35 @@ func getKeycloakEnv(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) []v1.EnvVar {
 		env = MergeEnvs(cr.Spec.KeycloakDeploymentSpec.Experimental.Env, env)
 	}
 
+	env = KeycloakSslEnvVariables(dbSecret, env)
+
 	return env
 }
 
-func KeycloakDeployment(cr *v1alpha1.Keycloak, dbSecret *v1.Secret) *v13.StatefulSet {
+func KeycloakSslEnvVariables(dbSecret *v1.Secret, env []v1.EnvVar) []v1.EnvVar {
+	if dbSecret != nil {
+		sslMode := string(dbSecret.Data[DatabaseSecretSslModeProperty])
+
+		if sslMode != "" {
+			dbParams := ""
+			// is the deployment already having JDBC_PARAMS set ?
+			for _, element := range env {
+				if strings.EqualFold(element.Name, KeycloakDatabaseConnectionParamsProperty) {
+					dbParams = element.Value + "&"
+					break
+				}
+			}
```

Comment:
I believe you don't need to use a `for` loops which is a bit cumbersome. You can do something like [this](https://stackoverflow.com/a/2050629) instead. But it's just a nitpick – I'll leave it up to you if you want to change it. ;)

Reply:
Yes, you're of course correct. I did not realize it's not a map. Sorry for the confusion!